packages/2d/src/components/CubicBezier.ts

```diff
@@ -0,0 +1,109 @@
+import {SignalValue} from '@motion-canvas/core/lib/signals';
+import {PossibleVector2, Vector2Signal} from '@motion-canvas/core/lib/types';
+import {CurveProps} from './Curve';
+import {CubicBezierSegment} from '../curves';
+import {PolynomialSegment} from '../curves/PolynomialSegment';
+import {computed, vector2Signal} from '../decorators';
+import {bezierCurveTo, lineTo, moveTo} from '../utils';
+import {Bezier, BezierOverlayInfo} from './Bezier';
+
+export interface CubicBezierProps extends CurveProps {
+  p0: SignalValue<PossibleVector2>;
+  p0X: SignalValue<number>;
+  p0Y: SignalValue<number>;
+
+  p1: SignalValue<PossibleVector2>;
+  p1X: SignalValue<number>;
+  p1Y: SignalValue<number>;
+
+  p2: SignalValue<PossibleVector2>;
+  p2X: SignalValue<number>;
+  p2Y: SignalValue<number>;
+
+  p3: SignalValue<PossibleVector2>;
+  p3X: SignalValue<number>;
+  p3Y: SignalValue<number>;
+}
+
+/**
+ * A node for drawing a cubic Bézier curve.
+ *
+ * @example
```

Comment:
The main example should use a `@preview` tag:
```suggestion
 * @preview
```
Unlike `@example`, it will be always expanded at the top of the documentation.